src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -97,7 +97,20 @@ protected SchemaTypes()
         /// </summary>
         /// <param name="schema">A schema for which this instance is created.</param>
         /// <param name="serviceProvider">A service provider used to resolve graph types.</param>
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
         public SchemaTypes(ISchema schema, IServiceProvider serviceProvider)
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+        {
+            Initialize(schema, serviceProvider);
+        }
+
+        /// <summary>
+        /// Initializes the instance for the specified schema, and with the specified type resolver.
+        /// </summary>
+        /// <param name="schema">A schema for which this instance is created.</param>
+        /// <param name="serviceProvider">A service provider used to resolve graph types.</param>
+        /// </summary>
+        protected virtual void Initialize(ISchema schema, IServiceProvider serviceProvider)
```

Comment:
1. virtual ?
2. Add some sort of check if `Initialize` has already been called.

Reply:
While I could remove `virtual`, as a matter of principle, all `protected` members should be `virtual`.  It doesn't really matter here, as you can run code in the constructor before or after `Initialize()`, or replace `Initialize()`.  Do you want me to remove `virtual` here?  (I plan to rewrite this entire class at some point anyway, hopefully before v5, but not sure if I have time.)

Reply:
I will add the check for Initialize.

Reply:
> Do you want me to remove virtual here?

Yes.

Reply:
Both changes done